### PR TITLE
Add compilation defaults for the BertMaskedLM task model 

### PR DESCRIPTION
**Pull request by ADITYADAS1999**
add fixes at https://github.com/keras-team/keras-nlp/issues/830

### CODE REVIEW

**Comment by soma2000-lang** [comment]
@ADITYADAS1999 I had already opened a pr 1 hour ago and clearly mentioned it in the isssue.So why did you open the same pr again?

**Comment by ADITYADAS1999** [comment]
> @ADITYADAS1999 I had already opened a pr 1 hour ago and clearly mentioned it in the isssue.So why did you open the same pr again?

maybe a typo ! Now it should fine 👍🏻

**Comment by mattdangerw** [comment]
Code looks good. Have we tried this out with a few different presets to make sure we are converging generally?

**Comment by ADITYADAS1999** [comment]
> Code looks good. Have we tried this out with a few different presets to make sure we are converging generally?

that's a bit tricky, in this case we have to working on some cases on convergence

File: keras_nlp/models/bert/bert_masked_lm.py [hunk 1]
```diff
@@ -139,6 +140,12 @@ def __init__(
         # All references to `self` below this line
         self.backbone = backbone
         self.preprocessor = preprocessor
+        self.compile(
+            loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+            optimizer=keras.optimizers.Adam(5e-5),
+            metrics=keras.metrics.SparseCategoricalAccuracy(),
```
Comment (mattdangerw) [inline keras_nlp/models/bert/bert_masked_lm.py:146]: I believe this should be `weighted_metrics` as we pass sample weights in the preprocessing.

Comment (ADITYADAS1999) [reply keras_nlp/models/bert/bert_masked_lm.py:146]: that's great 😀

**Comment by mattdangerw** [comment]
Thanks! Going to merge this in as another PR depends on it, with the understanding we might still need to tweak this learning rate after testing.

**Comment by ADITYADAS1999** [comment]
sounds great
